tests/conftest.py

```diff
@@ -250,3 +250,17 @@ def run(
         return stage
 
     return run
+
+
+@pytest.fixture
+def broken_rev(tmp_dir, scm, dvc):
```

Comment:
I don't think we need a broken rev and the whole functional test. Instead, just check that if the config file made the desired revision disappear from iter_revs with a unit test is enough.

Reply:
> disappear from iter_revs with a unit test is enough
I think it's not enough in this case.
If the underlying functionality of / gc / pull/ push` is changed (i.e. brancher usage is removed) there won't be any test indicating that the ignore_revs have stopped working for those commands.